backtrace-library/src/main/java/backtraceio/library/services/BacktraceHandlerThread.java

```diff
@@ -1,38 +1,78 @@
 package backtraceio.library.services;
 
+import android.content.Context;
 import android.os.Handler;
 import android.os.HandlerThread;
 import android.os.Looper;
 import android.os.Message;
 
 import java.util.List;
+import java.util.Map;
 
 import backtraceio.library.common.BacktraceSerializeHelper;
+import backtraceio.library.interfaces.Api;
 import backtraceio.library.logger.BacktraceLogger;
+import backtraceio.library.metrics.BacktraceHandlerInputEvents;
+import backtraceio.library.metrics.SummedEventsHandler;
+import backtraceio.library.metrics.SummedEventsPayload;
+import backtraceio.library.metrics.UniqueEventsHandler;
+import backtraceio.library.metrics.UniqueEventsPayload;
+import backtraceio.library.models.BacktraceMetricsSettings;
 import backtraceio.library.models.BacktraceResult;
 
 public class BacktraceHandlerThread extends HandlerThread {
 
+    private static transient String LOG_TAG = BacktraceHandlerThread.class.getSimpleName();
+
     private BacktraceHandler mHandler;
     private String url;
+    private UniqueEventsHandler mUniqueEventsHandler;
+    private SummedEventsHandler mSummedEventsHandler;
 
     BacktraceHandlerThread(String name, String url) {
         super(name);
         this.url = url;
         this.start();
     }
 
+    UniqueEventsHandler createUniqueEventsHandler(Context context, Map<String, Object> customAttributes, Api api, BacktraceMetricsSettings settings) {
+        this.mUniqueEventsHandler = new UniqueEventsHandler(context, customAttributes, api, this, settings);
+        return mUniqueEventsHandler;
+    }
+
+    SummedEventsHandler createSummedEventsHandler(Context context, Map<String, Object> customAttributes, Api api, BacktraceMetricsSettings settings) {
+        this.mSummedEventsHandler = new SummedEventsHandler(context, customAttributes, api, this, settings);
+        return mSummedEventsHandler;
+    }
+
     @Override
     protected void onLooperPrepared() {
         super.onLooperPrepared();
         mHandler = new BacktraceHandler(this.getLooper(), this.url);
     }
 
-    void sendReport(BacktraceHandlerInput data) {
+    void sendReport(BacktraceHandlerInputReport data) {
+        send(data);
+    }
 
+    void send(BacktraceHandlerInput data) {
         Message message = new Message();
         message.obj = data;
-        mHandler.sendMessage(message);
+        if (data instanceof BacktraceHandlerInputReport) {
+            mHandler.sendMessage(message);
+        } else if (data instanceof BacktraceHandlerInputEvents) {
```

Comment:
mmm these instanceof things make me queasy. Can't we use inheritance here? Have a single super handler that has send methods for all types and figures our where to route it?